WordPress/Classes/ViewRelated/NewGutenberg/NewGutenbergViewController.swift

```diff
@@ -316,6 +327,70 @@ class NewGutenbergViewController: UIViewController, PostEditor, PublishingEditor
             WordPressAppDelegate.crashLogging?.logJavaScriptException(exception, callback: callback)
         }
     }
+
+    // MARK: - Activity Indicator
+
+    private func showActivityIndicator() {
+        let indicator = UIActivityIndicatorView(style: .large)
+        indicator.color = .gray
+        indicator.translatesAutoresizingMaskIntoConstraints = false
+        view.addSubview(indicator)
+
+        NSLayoutConstraint.activate([
+            indicator.centerXAnchor.constraint(equalTo: view.centerXAnchor),
+            indicator.centerYAnchor.constraint(equalTo: view.centerYAnchor)
+        ])
+
+        indicator.startAnimating()
+        self.activityIndicator = indicator
+    }
+
+    private func hideActivityIndicator() {
+        activityIndicator?.stopAnimating()
+        activityIndicator?.removeFromSuperview()
+        activityIndicator = nil
+    }
+
+    // MARK: - Block Editor Settings
+
+    private func fetchBlockEditorSettings() {
+        guard let service = rawBlockEditorSettingsService else {
+            startEditor()
+            return
+        }
+
+        Task { @MainActor in
+            // Start the editor with default settings after 3 seconds
+            let timeoutTask = Task {
```

Comment:
This is a clever way to address this – I think ideally we'd just use the `timeout` property on `URLRequest`, but we don't _really_ own `WordPressOrgRestApi`. I guess we'll come back to this when it's time to adopt the Rust-layer REST API, and we can switch it over then.